snowfakery/data_generator_runtime_object_model.py

```diff
@@ -80,6 +80,37 @@ def execute(
         interp.register_variable(name, value)
 
 
+class ForEachVariableDefinition:
+    varname: str
+    expression: "StructuredValue"
+
+    def __init__(
+        self, filename: str, line_num: int, varname: str, expression: Definition
+    ):
+        self.varname = varname
+        self.expression = expression
+        self.filename = filename
+        self.line_num = line_num
+
+    def evaluate(self, context: RuntimeContext) -> FieldValue:
+        """Evaluate the expression"""
```

Comment:
It both does the evaluation and sets the `recalculate_every_time` flag to true for the given context. Is that side effect worth mentioning in the docstring?

Reply:
https://github.com/SFDO-Tooling/Snowfakery/pull/457/commits/b7054d2b2adf278da6ba5fbcc1be10ebcb9aa76a#diff-2bba575d2df68a6e02fb6177a24fc661bcf343369942fdf8848b644612345a56R84